drafts/Asynchronous-programming-made-resilient-with-MPFT.adoc

```diff
@@ -0,0 +1,230 @@
+
+==== JAX-RS client API `CompletionStage` example:
+
+[source,java]
+----
+private Client client = ClientBuilder.newClient();
+
+public CompletionStage<String> clientDemo() {
+
+	CompletionStage<String> response = client.target("http://example.com/resource")
+					.request(MediaType.TEXT_PLAIN)
+					.rx()
+					.get(String.class);
+
+	return response;
+
+}
+----
+==== Fetching the following resource:
+[source,java]
+----
+@GET
+@Path("/resource")
+public CompletionStage<String> fetchExample() { 
+
+	return CompletableFuture.completedFuture("responseText");
+
+}
+----
+
+If we call the `clientDemo()` method, without any annotations, it works as we expect. We call it, receive a `CompletionStage<String>` (named `response` in the example), and then we can add an action to take when the `CompletionStage` completes.
+
+[source,java]
+----
+response.thenAccept(System.out::println);
+--> responseText
+----
+
+However, what happens if we want to use the `@Retry` annotation with our method? The answer is nothing. 
+
+Even if the HTTP request fails, the request doesn't get retried. Why is this? Well, Fault Tolerance acts around method calls. Normally, if you annotate a method with `@Retry` and it throws an exception then it gets retried. However, when we do an HTTP request through this JAX-RS client API, if there's a problem it doesn't throw an exception. It can't throw an exception. It's going to do the request asynchronously, so when this method returns, it probably hasn't even made the request yet, let alone found out that it failed. If an exception does occur, it will be propagated  to the `CompletionStage` and can be handled there. But the result is, that this method will never throw an exception, even if the request fails and so the request will never be retried.
+
+*@Asynchronous to the rescue!*
+
+However, if we add the `@Asynchronous` annotation and the method returns a `CompletionStage`, then the Fault Tolerance logic gets applied when the `CompletionStage` completes, rather than when the method returns. 
+
+[source,java]
+----
+@Asynchronous
+@Retry
+public CompletionStage<String> clientDemo() {
+
+	...
+
+}
+----
+
+That means that when we call this method and it returns a `CompletionStage`, it's only when that `CompletionStage` completes that Fault Tolerance will look at the result and decide whether to retry. So, if the request fails, the `CompletionStage` completes with an exception, Fault Tolerance decides that a Retry is needed and it calls the method again. As before, Fault Tolerance has intercepted the method call, so the `CompletionStage` returned to the caller is a different `CompletionStage` so that the caller doesn't get the result until all the Retries have been completed.
+
+So, to recap, to use Fault Tolerance with an asynchronous API you must:
+
+- **Return a `CompletionStage` from your method** - You can't do this with a `Future`, it must be a `CompletionStage`.
```

Comment:
The first part of this article does not mention anything about Future not applicable. You should explain why prior to this recap.